### ISSUE TITLE
[MASSEMBLY-567] Version wildcard in dependencySet include does not work

### ISSUE THREAD

**Issue by jira-importer** [apache/maven-assembly-plugin#737]
**[Stephan Oudmaijer](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=soudmaijer)** opened **[MASSEMBLY-567](https://issues.apache.org/jira/browse/MASSEMBLY-567?redirect=false)** and commented

In our assembly descriptor we use a version wildcard for a dependencySet include, see the following snippet:

\<dependencySet>
\<outputDirectory>./configuration\</outputDirectory>
\<includes>
\<include>com.xyz.admin:xyz-admin-tool-config:*:zip:config\</include>
\</includes>
\<unpack>true\</unpack>
\</dependencySet>

This used to work in version 2.2-beta-5 but since 2.2 but also in 2.2.1 this does not work anymore. The version wildcard causes problems. According to the documentation:

Artifacts are matched by a set of identifier strings. In the following strings, type is 'jar' by default, and classifier is omitted if null.

groupId:artifactId:version:type:classifier ( artifact.getId() )

Any '*' character in an include/exclude pattern will result in the pattern being split, and the sub-patterns being matched within the three artifact identifiers mentioned above, using String.indexOf(..).

I suspect this is a bug? Or the behavior has changed, but this is not in-line with the documentation.

---

**Affects:** 2.2, 2.2.1

**Issue Links:**
- [MASSEMBLY-599](https://issues.apache.org/jira/browse/MASSEMBLY-599) Mistake in javadoc for dependencySet/includes

**Comment by jira-importer** [comment]
**[Benson Margulies](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=bmargulies)** commented

It's not just wildcards that are broken, even putting the correct version into place manually fails.

**Comment by jira-importer** [comment]
**[Benson Margulies](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=bmargulies)** commented

As far as I can tell, the spec order has always been the following, which is what is in the doc. I just tried it (with the current snapshot) and it worked fine, with a * in version.

groupId:artifactId:type:version[:classifier]

**Comment by jira-importer** [comment]
**[Dennis Lundberg](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=dennisl@apache.org)** commented

The documentation was wrong. The correct format is:
groupId:artifactId:type[:classifier]:version

Fixed in [r1402850](http://svn.apache.org/viewvc?view=revision&revision=1402850).

**Comment by jira-importer** [comment]
**[Benson Margulies](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=bmargulies)** commented

In other words, it would seem that this change breaks the builds of everyone who had followed the doc. Are you sure you want to do that?

**Comment by jira-importer** [comment]
**[Dennis Lundberg](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=dennisl@apache.org)** commented

What change are you referring to?
All I did was correct the documentation, so that it is in sync with reality.

**Comment by jira-importer** [comment]
**[Benson Margulies](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=bmargulies)** commented

Dennis, could there have been two different places in the doc with two different specs? Either that, or I was apparently drunk.

**Comment by jira-importer** [comment]
**[Dennis Lundberg](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=dennisl@apache.org)** commented

Benson,
In short - the docs were a mess. There were at least 3 different specs. Look at the issues that was resolved as part of the commit and follow the links in those issues to see the different specs in the current docs. What I did was align them so that they all tell the same story - the correct story.
